packages/common/src/controllers/AccountController.ts

```diff
@@ -188,20 +166,27 @@ export default class AccountController {
   };
 
   login = async (email: string, password: string, referrer: string) => {
-    const { config, accessModel } = useConfigStore.getState();
-
     useAccountStore.setState({ loading: true });
 
-    const response = await this.accountService.login({ config, email, password, referrer });
+    try {
+      const response = await this.accountService.login({ email, password, referrer });
 
-    if (response) {
-      await this.afterLogin(response.user, response.customerConsents, accessModel);
+      if (!response) throw new Error("Couldn't login");
```

Comment:
Can we refactor this to avoid throwing errors to control flow? Maybe leave the `if (response)` check and move the return statement up to the end of the try block. Then let the catch block catch and rethrow for the known special cases, and finally let the flow cascade down to the end of the method where you can throw the `wrong_combination` error if nothing else has happened.

Reply:
That would look something like this?

- Only rethrow known errors (wrong_email)
- Fallback to wrong_combination for all unexpected errors
- Use finally to always clear the loading state (this also works with the `return` in the `try` block

```ts
  login = async (email: string, password: string, referrer: string) => {
    useAccountStore.setState({ loading: true });

    try {
      const response = await this.accountService.login({ email, password, referrer });
      
      if (response) {
        await this.afterLogin(response.user, response.customerConsents);
        await this.favoritesController?.restoreFavorites().catch(logDev);
        await this.watchHistoryController?.restoreWatchHistory().catch(logDev);
        
        return;
      }
    } catch (error: unknown) {
      if (error instanceof Error && error.message.toLowerCase().includes('invalid param email')) {
        throw new FormValidationError({ email: [i18next.t('account:login.wrong_email')] });
      }
    } finally {
      useAccountStore.setState({ loading: false });
    }

    throw new FormValidationError({ form: [i18next.t('account:login.wrong_combination')] });
  };
```